go/ql/lib/semmle/go/frameworks/Afero.qll

```diff
@@ -0,0 +1,112 @@
+          ["Create", "Open", "OpenFile", "Remove", "RemoveAll", "Mkdir", "MkdirAll"]) and
+        this = m.getACall()
+        or
+        m.hasQualifiedName(aferoPackage(), "ReadOnlyFs",
+          ["Create", "Open", "OpenFile", "ReadDir", "ReadlinkIfPossible", "Mkdir", "MkdirAll"]) and
+        this = m.getACall()
+        or
+        m.hasQualifiedName(aferoPackage(), "OsFs",
+          [
+            "Create", "Open", "OpenFile", "ReadlinkIfPossible", "Remove", "RemoveAll", "Mkdir",
+            "MkdirAll"
+          ]) and
+        this = m.getACall()
+        or
+        m.hasQualifiedName(aferoPackage(), "MemMapFs",
+          ["Create", "Open", "OpenFile", "Remove", "RemoveAll", "Mkdir", "MkdirAll"]) and
+        this = m.getACall()
+      )
+    }
+
+    override DataFlow::Node getAPathArgument() { result = this.getArgument(0) }
+  }
+
+  /**
+   * The File system access sinks of [afero](https://github.com/spf13/afero) framework utility functions
+   *
+   * Afero Type is basically is an wrapper around utility functions which make them like a method, look at [here](https://github.com/spf13/afero/blob/cf95922e71986c0116204b6eeb3b345a01ffd842/ioutil.go#L61)
+   *
+   * The Types that are not vulnerable: `afero.BasePathFs` and `afero.IOFS`
+   */
+  class AferoUtilityFunctionSystemAccess extends FileSystemAccess::Range, DataFlow::CallNode {
+    int pathArg;
+
+    AferoUtilityFunctionSystemAccess() {
+      // utility functions
+      exists(Function f |
+        f.hasQualifiedName(aferoPackage(),
+          ["WriteReader", "SafeWriteReader", "WriteFile", "ReadFile", "ReadDir"]) and
+        this = f.getACall() and
+        pathArg = 1 and
+        not aferoSanitizer(this.getArgument(0))
+      )
+      or
+      exists(Method m |
+        m.hasQualifiedName(aferoPackage(), "Afero",
+          ["WriteReader", "SafeWriteReader", "WriteFile", "ReadFile", "ReadDir"]) and
+        this = m.getACall() and
+        pathArg = 0 and
+        not aferoSanitizer(this.getReceiver())
+      )
+    }
+
+    override DataFlow::Node getAPathArgument() { result = this.getArgument(pathArg) }
+  }
+
+  /**
+   * Holds if the Afero utility function has a first argument of a safe type like `NewBasePathFs`.
+   *
+   * e.g.
+   * ```
+   * basePathFs := afero.NewBasePathFs(osFS, "tmp")
+   * afero.ReadFile(basePathFs, filepath)
+   * ```
+   */
+  predicate aferoSanitizer(DataFlow::Node n) {
+    exists(Function f |
+      f.hasQualifiedName(aferoPackage(), ["NewBasePathFs", "NewIOFS"]) and
+      DataFlow::localFlow(f.getACall(), n)
+    )
+  }
+
+  /**
+   * Holds if there is a dataflow node from n1 to n2 when initializing the Afero instance
+   *
+   * A helper for `aferoSanitizer` for when the Afero instance is initialized with one of the safe FS types like IOFS
+   *
+   * e.g.`n2 := &afero.Afero{Fs: afero.NewBasePathFs(osFS, "./")}` n1 is `afero.NewBasePathFs(osFS, "./")`
+   */
+  predicate additionalTaintStep(DataFlow::Node n1, DataFlow::Node n2) {
```

Comment:
If I understand correctly, you need this to be included in `isAdditionalFlowStep` for flow configurations that relate to Afero file system accesses. There is a better way of doing this: extend the class `TaintTracking::AdditionalTaintStep`. You can see examples of this [here](https://github.com/amammad/codeql/blob/0f5dd40ff16ba35524701510dcb0fbb67636a5a1/go/ql/lib/semmle/go/frameworks/Protobuf.qll#L65). Then this step will be included in all taint tracking flows. Hopefully this means your test will start working and you can remove the `SPURIOUS` label